test/blogs_api_web/controllers/post_controller_test.exs

```diff
@@ -6,27 +6,179 @@ defmodule BlogsApiWeb.PostControllerTest do
   import BlogsApiWeb.Auth.Guardian
   use BlogsApiWeb.ConnCase, async: true
 
-  describe "delete" do
-    setup %{conn: conn} do
-      params = %{
-        display_name: "Harry Potter",
-        password: "123456",
-        email: "harry@email.com",
-        image:
-          "https://ogimg.infoglobo.com.br/in/24440303-24f-31c/FT1086A/87996533_SCAtor-Daniel-Redcliff-como-Harry-Potter.-Foto-Divulgacao.jpg"
-      }
+  @params %{
+    display_name: "Harry Potter",
+    password: "123456",
+    email: "harry@email.com",
+    image:
+      "https://ogimg.infoglobo.com.br/in/24440303-24f-31c/FT1086A/87996533_SCAtor-Daniel-Redcliff-como-Harry-Potter.-Foto-Divulgacao.jpg"
+  }
+
+  @create_attrs %{content: "some content", title: "some title"}
+
+  @update_attrs %{content: "Conteúdo atualizado", title: "titulo"}
 
-      {:ok, user} = BlogsApi.create_user(params)
-      {:ok, token, _claims} = encode_and_sign(user)
+  defp setup_user(%{conn: conn}) do
+    {:ok, user} = BlogsApi.create_user(@params)
+    {:ok, token, _claims} = encode_and_sign(user)
 
-      conn = put_req_header(conn, "authorization", "Bearer #{token}")
-      {:ok, conn: conn}
+    conn = put_req_header(conn, "authorization", "Bearer #{token}")
+    {:ok, conn: conn}
+    %{conn: conn, user: user}
+  end
+
+  defp setup_post(%{conn: conn, user: user}) do
+    {:ok, post} = BlogsApi.create_post_user(user, @create_attrs)
+    {:ok, conn: conn, user: user, post: post}
+  end
+
+  describe "index/1" do
+    setup :setup_user
+
+    test "Retorna todos os posts", %{conn: conn} do
+      response =
+        get(conn, Routes.post_path(conn, :index))
+        |> json_response(:ok)
+
+      assert response
     end
+  end
+
+  describe "show/2" do
+    setup :setup_user
+    setup :setup_post
+
+    test "Retorna um post, quando o post é encontrado", %{conn: conn, user: _user, post: post} do
+      connection = get(conn, Routes.post_path(conn, :show, post.id))
+
+      assert %{
+          "id" => _id,
+          "title" => "some title",
+          "content" => "some content",
+          "published" => _published,
+          "updated" => _updated,
+          "user" => %{
+            "id" => _user_id,
+            "display_name" => "Harry Potter",
+            "email" => "harry@email.com",
+            "image" =>
+            "https://ogimg.infoglobo.com.br/in/24440303-24f-31c/FT1086A/87996533_SCAtor-Daniel-Redcliff-como-Harry-Potter.-Foto-Divulgacao.jpg"
+          }
+      } = json_response(connection, :ok)
```

Comment:
Acho que aqui, se você não vai conferir os valores que estão com `_` no início, não precisa colocar no pattern match.

```suggestion
      assert %{
          "title" => "some title",
          "content" => "some content",
          "user" => %{
            "display_name" => "Harry Potter",
            "email" => "harry@email.com",
            "image" =>
            "https://ogimg.infoglobo.com.br/in/24440303-24f-31c/FT1086A/87996533_SCAtor-Daniel-Redcliff-como-Harry-Potter.-Foto-Divulgacao.jpg"
          }
      } = json_response(connection, :ok)
```